### PR TITLE
Feat/signin

### PR DESCRIPTION
**Pull request by saransh7**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -35,6 +35,15 @@
     android:icon="@mipmap/ic_launcher"
     android:label="@string/app_name"
     android:theme="@style/AppTheme">
+      <activity
```
Comment (ShikherVerma) [inline app/src/main/AndroidManifest.xml]: code style not consistent

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 2]
```diff
@@ -1,38 +1,50 @@
...
-import android.widget.AdapterView;
 
-import com.mikepenz.iconics.typeface.FontAwesome;
-import com.mikepenz.materialdrawer.Drawer;
-import com.mikepenz.materialdrawer.DrawerBuilder;
-import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
-import com.mikepenz.materialdrawer.model.interfaces.IDrawerItem;
-import com.mikepenz.materialdrawer.model.interfaces.Nameable;
-import com.mikepenz.materialdrawer.util.KeyboardUtil;
+        import android.content.Intent;
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: why this extra space ?

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 3]
```diff
@@ -1,38 +1,50 @@
...
+        import com.mikepenz.materialdrawer.model.interfaces.Nameable;
+        import com.mikepenz.materialdrawer.util.KeyboardUtil;
 
-import in.antaragni.ant.fragments.AboutFragment;
-import in.antaragni.ant.fragments.ContactFragment;
-import in.antaragni.ant.fragments.EventFragment;
-import in.antaragni.ant.fragments.HomeFragment;
-import in.antaragni.ant.fragments.MapFragment;
-import in.antaragni.ant.fragments.ScheduleFragment;
+        import in.antaragni.ant.fragments.AboutFragment;
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: why extra space ?

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 4]
```diff
@@ -1,38 +1,50 @@
...
+        import in.antaragni.ant.fragments.ContactFragment;
+        import in.antaragni.ant.fragments.EventFragment;
+        import in.antaragni.ant.fragments.HomeFragment;
+        import in.antaragni.ant.fragments.MapFragment;
+        import in.antaragni.ant.fragments.ScheduleFragment;
 
-public class MainActivity extends AppCompatActivity
-{
+public class MainActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener
+{ private GoogleApiClient CLIENT;
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: code style not consistent

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 5]
```diff
@@ -61,12 +80,15 @@ protected void onCreate(Bundle savedInstanceState)
...
       .addDrawerItems(
+
         new PrimaryDrawerItem().withName(R.string.drawer_item_home).withIcon(FontAwesome.Icon.faw_home).withIdentifier(HOME),
         new PrimaryDrawerItem().withName(R.string.drawer_item_schedule).withIcon(FontAwesome.Icon.faw_calendar).withIdentifier(SCHEDULE),
         new PrimaryDrawerItem().withName(R.string.drawer_item_events).withIcon(FontAwesome.Icon.faw_eye).withIdentifier(EVENTS),
         new PrimaryDrawerItem().withName(R.string.drawer_item_maps).withIcon(FontAwesome.Icon.faw_map_marker).withIdentifier(MAP),
         new PrimaryDrawerItem().withName(R.string.drawer_item_contact).withIcon(FontAwesome.Icon.faw_users).withIdentifier(CONTACT),
-        new PrimaryDrawerItem().withName(R.string.drawer_item_about).withIcon(FontAwesome.Icon.faw_book).withIdentifier(ABOUT))
+        new PrimaryDrawerItem().withName(R.string.drawer_item_about).withIcon(FontAwesome.Icon.faw_book).withIdentifier(ABOUT),
+              new PrimaryDrawerItem().withName("SignOut").withIcon(FontAwesome.Icon.faw_sign_out).withIdentifier(SIGNOUT),
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: code style not consistent. And please use `string.xml` Don't write "SignOut".

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1,30 +1,74 @@
...
-  <!-- TODO: Remove or change this placeholder text -->
-  <string name="hello_blank_fragment">Hello blank fragment</string>
-  <string name="title_activity_maps">Map</string>
-  <string name="title_activity_event_detail">EventDetailActivity</string>
-  <string name="hello_world">Hello world!</string>
-
-  <!--
+
+
+    <string name="title_activity_main">MainActivity</string>
```
Comment (ShikherVerma) [inline app/src/main/res/values/strings.xml:4]: too many extra white spaces.

**Comment by ShikherVerma** [comment]
@saransh7 @santwanaverma15 @piyush-ag @aakanksha287 Please take a note: 
Go to your `Editor Settings` and change all spaces to 2.  
Run `reformat code` option in `Code` tab before commiting. Algo:
- Choose the `java` folder in sidebar
- Select `Only VCS changed text`, `Optimize Imports`, `Include Sub directories`
- Choose the `res` folder in sidebar
- Select all options

**Comment by ShikherVerma** [comment]
Since I have merged a pull request. plz `rebase` from master.

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 7]
```diff
@@ -1,15 +1,29 @@
 package in.antaragni.ant;
-
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java:2]: why delete this line ?

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 8]
```diff
@@ -25,17 +39,19 @@
...
   protected static int SCHEDULE = 2;
   protected static int EVENTS = 3;
   protected static int MAP = 4;
   protected static int CONTACT = 5;
   protected static int ABOUT = 6;
+  protected static int SIGNOUT=7;
 
   //save our header or result
   private Drawer result = null;
+  private GoogleApiClient CLIENT;
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: @saransh7 @santwanaverma15 @Piyush-ag plz note. ALLCAPS are used for constants. plz rename this one.

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 9]
```diff
@@ -25,17 +39,19 @@
 import in.antaragni.ant.fragments.MapFragment;
 import in.antaragni.ant.fragments.ScheduleFragment;
 
-public class MainActivity extends AppCompatActivity
+public class MainActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener
 {
   protected static int HOME = 1;
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: Make these Caps variables `final` as they are constants

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 10]
```diff
@@ -61,13 +83,16 @@ protected void onCreate(Bundle savedInstanceState)
...
-      .withOnDrawerItemClickListener(new Drawer.OnDrawerItemClickListener()
+
+              new PrimaryDrawerItem().withName(R.string.drawer_item_home).withIcon(FontAwesome.Icon.faw_home).withIdentifier(HOME),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_schedule).withIcon(FontAwesome.Icon.faw_calendar).withIdentifier(SCHEDULE),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_events).withIcon(FontAwesome.Icon.faw_eye).withIdentifier(EVENTS),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_maps).withIcon(FontAwesome.Icon.faw_map_marker).withIdentifier(MAP),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_contact).withIcon(FontAwesome.Icon.faw_users).withIdentifier(CONTACT),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_about).withIcon(FontAwesome.Icon.faw_book).withIdentifier(ABOUT),
+              new PrimaryDrawerItem().withName(R.string.drawer_item_signout).withIcon(FontAwesome.Icon.faw_sign_out).withIdentifier(SIGNOUT),
+              new PrimaryDrawerItem().withName(getIntent().getExtras().getString("username")))
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java]: why is `"username"` here ? If its a UI visible string then move it to `string.xml` and if its a code related contant then move it to the top of the class and make it `final static`. Also if u are using this same key (ie. "username") somewhere else to load data to intent then there too, refer to the static variable of this class.

File: app/src/main/java/in/antaragni/ant/MainActivity.java [hunk 11]
```diff
@@ -61,13 +83,16 @@ protected void onCreate(Bundle savedInstanceState)
       .withTranslucentStatusBar(false)
       .withActionBarDrawerToggleAnimated(true)
       .addDrawerItems(
-        new PrimaryDrawerItem().withName(R.string.drawer_item_home).withIcon(FontAwesome.Icon.faw_home).withIdentifier(HOME),
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/MainActivity.java:64]: why add new white space? if it was added by reformatting then fine. But if u have added it manually, remove them

File: app/src/main/java/in/antaragni/ant/SignInActivity.java [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+    // [START signIn]
+    private void signIn() {
+        Intent signInIntent = Auth.GoogleSignInApi.getSignInIntent(mGoogleApiClient);
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+    // [END signIn]
+
+
+    public void init() {
+        Intent toy = new Intent(SignInActivity.this, MainActivity.class);
```
Comment (ShikherVerma) [inline app/src/main/java/in/antaragni/ant/SignInActivity.java:157]: `toy` u should use `i` for generic intent name or what ever the IDE suggest. Also the keys "username" and "image" should be `static final` and declared at the start of the class

File: app/src/main/res/layout/activity_main.xml [hunk 13]
```diff
@@ -26,4 +29,5 @@
       android:layout_height="match_parent"/>
   </FrameLayout>
 
+
```
Comment (ShikherVerma) [inline app/src/main/res/layout/activity_main.xml:30]: farji extra whitespace

File: app/src/main/res/layout/activity_sign_in.xml [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        <ImageView
+            android:id="@+id/google_icon"
+            android:layout_width="70dp"
+            android:layout_height="70dp"
+            android:layout_marginBottom="10dp"
+            android:layout_marginTop="@dimen/g_top_margin"
+            android:contentDescription="@string/desc_google_icon"
+            android:src="@drawable/googleg_color" />
+
+
```
Comment (ShikherVerma) [inline app/src/main/res/layout/activity_sign_in.xml]: extra white space. did u run reformat as I told u to ?

File: app/src/main/res/layout/activity_sign_in.xml [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+            android:fadeScrollbars="true"
+            android:gravity="center"
+            android:maxLines="5"
+            android:padding="10dp"
+            android:scrollbars="vertical"
+            android:textColor="@android:color/white"
+            android:textSize="14sp" />
+
+    </LinearLayout>
+
```
Comment (ShikherVerma) [inline app/src/main/res/layout/activity_sign_in.xml]: extra white space

File: app/src/main/res/values/dimens.xml [hunk 16]
```diff
@@ -1,29 +1,30 @@
 <resources>
-  <!-- Default screen margins, per the Android Design guidelines. -->
```
Comment (ShikherVerma) [inline app/src/main/res/values/dimens.xml:2]: why remove spaces here ? was it done by auto formatting ?

File: app/src/main/res/values-w820dp/dimens.xml [hunk 17]
```diff
@@ -1,6 +1,6 @@
 <resources>
-  <!-- Example customization of dimensions originally defined in res/values/dimens.xml
+    <!-- Example customization of dimensions originally defined in res/values/dimens.xml
```
Comment (ShikherVerma) [inline app/src/main/res/values-w820dp/dimens.xml:2]: why remove spaces here ? was it done by auto formatting ?

Comment (saransh7) [inline app/src/main/res/values-w820dp/dimens.xml:2]: Yep it was done by auto  formatting

File: app/src/main/res/values/dimens.xml (hunk 16 above)
Comment (saransh7) [inline app/src/main/res/values/dimens.xml:2]: Yep

**Pull request closed without merging**